pixeltable/func/function.py

```diff
@@ -123,11 +124,11 @@ def help_str(self) -> str:
     def __call__(self, *args: Any, **kwargs: Any) -> 'pxt.exprs.FunctionCall':
         from pixeltable import exprs
 
-        mono_fn, bound_args = self._bind_to_matching_signature(args, kwargs)
-        return_type = mono_fn.call_return_type(args, kwargs)
-        return exprs.FunctionCall(mono_fn, bound_args, return_type)
+        resolved_fn, bound_args = self._bind_to_matching_signature(args, kwargs)
+        return_type = resolved_fn.call_return_type(args, kwargs)
+        return exprs.FunctionCall(resolved_fn, bound_args, return_type)
 
-    def _bind_to_matching_signature(self, args: Sequence[Any], kwargs: dict[str, Any]) -> tuple[Function, dict[str, Any]]:
+    def _bind_to_matching_signature(self, args: Sequence[Any], kwargs: dict[str, Any]) -> tuple[Self, dict[str, Any]]:
```

Comment:
I would still argue that this is the actual overload resolution, whereas _overload_resolution() is basically get_overload(idx). Instead of the current _overload_resolution(), you could even do a
@property
def overloads() -> list[Function]
    if len(_resolutions) == 0:
        ...
    return self._resolutions
and then fn.overloads[i] looks pretty compact and still doesn't obscure anything.

Reply:
Hmm. In my mind they're "resolutions", not "overloads": the subtle difference being that the "overloads" are the different signatures and bodies that belong to the polymorphic function, and the "resolutions" are the corresponding signature-specific (non-polymorphic) functions.
<!-- Reviewable comment -OEWklgsBWQssmWjaBC0:b3mffto -->
<!-- Sent from Reviewable.io -->